modules/ROOT/pages/upgrade-index.adoc

```diff
@@ -1,35 +1,282 @@
+```
+... Paste the code snippet in a text editor for reference. 
+.. Start the upgrade procedure:
+... Using your terminal open on a controller node, run the following command, substituting the value of the <cluster-installer-url> parameter with the *Installer package* URL value: 
++
+----
+sudo ./rtfctl appliance upgrade --url <cluster-installer-url>
+----
++
+This command also supports using the `--file` parameter to reference an installer package already downloaded on the node.
++
+A confirmation is displayed that indicates the upgrade is being performed in the background.
++
+.. Run `sudo gravity status` on a node and verify that the cluster status is “updating”.
+.. Monitor the upgrade procedure progress:
+... Using your terminal on a controller node, run the following command: 
++
+----
+sudo gravity plan
+----
++
+The upgrade can take more than an hour to complete.
++
+The Runtime Fabric and Runtime Manager connection often disconnects multiple times during the upgrade procedure. Connectivity is restored after the upgrade completes.
++
+The upgrade procedure is designed to keep deployed applications running and available to serve requests. The upgrade is applied in a rolling method:
++
+.... A node is selected by the procedure to apply the upgrade.
+.... Applications running on the selected node are redeployed to another node.
+.... The selected node is removed from the cluster.
+.... The upgrade is applied on the selected node.
+.... The selected node is added to the cluster after upgrading.
+
+.. Confirm the upgrade has completed successfully:
+... Run `sudo gravity status` and verify that the cluster status transitioned from *Updating* to *Active*.
+.. If the Runtime Fabric cluster version was 1.0.x prior to upgrading, and an HTTP proxy is in use, run the following command to apply the HTTP proxy settings: 
++
+----
+sudo ./rtfctl apply http-proxy --confirm existing
+----
+.. Perform the following step on *every node* to make sure system configurations are up-to-date:
+
+... Open a terminal to your Runtime Fabric controller/worker node.
+... If needed, download the latest `rtfctl` command-line utility:
++
+----
+cd /opt/anypoint/runtimefabric
+curl -L https://anypoint.mulesoft.com/runtimefabric/api/download/rtfctl/latest -o rtfctl
+----
++
+... Change file permissions for the `rtfctl` binary: 
++
+----
+chmod +x rtfctl
+----
++
+... Run the `apply system-configurations` command in `rtfctl`:
++
+----
+sudo ./rtfctl apply system-configuration 
+----
+
+== Resume a Cluster-level Component (Appliance) Upgrade
+
+If a cluster-level (installer package) components upgrade step fails, try to resume the upgrade. 
+
+Resumed upgrades are attached to your terminal session, so ensure you have a stable connection before attempting to resume an upgrade.
+
+. On the terminal open to the controller node that was used to start the upgrade, navigate to the directory with the installer bundle files, as shown in the following example:
```

Comment:
Hmm...I might have misunderstood all the previous wordings around open terminals and controller nodes. Would you please review this one and the previous and ensure that they clearly convey the relationship between terminal and node?